include/cutlass/gemm/kernel/xe_gemm.hpp

```diff
@@ -153,6 +153,14 @@ class GemmUniversal<
 
   static bool
   can_implement(Arguments const& args) {
+    auto m = get<0>(args.problem_shape);
+    auto n = get<1>(args.problem_shape);
+    auto k = get<2>(args.problem_shape);
+    bool m_valid = m > 0;
+    bool n_valid = n > 0 && n % 4 == 0;
+    bool k_valid = k > 0 && k % get<2>(TileShape{}) == 0;
+    if (!(m_valid and n_valid and k_valid)) return false;
```

Comment:
Total nit so please feel free to ignore:
```suggestion
    shape_implementable =  (m_valid and n_valid and k_valid);
```
& then add it to the return statement. Slightly neater (?) but with potentially redundant checks which your solution short-circuits.